automaton/src/Data/Automaton/Filter.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DerivingVia #-}
+
+module Data.Automaton.Filter where
```

Comment:
Actually I should implement this for `Stream` first, and then pull through to `Automaton`